store/items.ts

```diff
@@ -0,0 +1,71 @@
+import Vue from 'vue';
+import { BaseItemDto } from '@jellyfin/client-axios';
+import { ActionTree, GetterTree, MutationTree } from 'vuex';
+import map from 'lodash/map';
+import some from 'lodash/some';
+import forEach from 'lodash/forEach';
+
+export interface ItemsState {
+  byId: Record<string, BaseItemDto>;
+  allIds: string[];
+}
+
+export const defaultState = (): ItemsState => ({
+  byId: {},
+  allIds: []
+});
+
+export const state = defaultState;
+
+export const getters: GetterTree<ItemsState, ItemsState> = {
+  getItem: (state) => (id: string): BaseItemDto | undefined => state.byId[id],
+  getItems: (state) => (ids: string[]): BaseItemDto[] =>
+    map(ids, (id) => state.byId[id])
+};
+
+export const mutations: MutationTree<ItemsState> = {
+  ADD_ITEM(state: ItemsState, { item }: { item: BaseItemDto }) {
+    if (!item.Id) {
+      throw new Error('No item ID');
+    }
+    Vue.set(state.byId, item.Id, item);
+    if (!state.allIds.includes(item.Id)) {
+      state.allIds.push(item.Id);
+    }
+  },
+  DELETE_ITEM(state: ItemsState, { id }: { id: string }) {
+    delete state.byId[id];
+    const idx = state.allIds.indexOf(id);
+    if (idx > -1) {
```

Comment:
```suggestion
    if (idx !== undefined) {
```

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf#return_value though?

Reply:
Could've said that I got an ``undefined`` from this function the other day. Good to know!